versioned_docs/version-2021fa/lecture10.md

```diff
@@ -0,0 +1,159 @@
+      <TaskList />
+      <StatusBar style="auto" />
+    </View>
+  );
+}
+
+// how to style elements
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: '#fff',
+    alignItems: 'center',
+    justifyContent: 'center',
+    gap: '2rem',
+  },
+});
+```
+
+```tsx title=Playlist.tsx
+import {
+  Button,
+  FlatList,
+  Text,
+  TextInput,
+  View,
+  StyleSheet,
+} from 'react-native';
+import React, { useState } from 'react';
+
+const styles = StyleSheet.create({
+  // space out the container a bit
+  container: { display: 'flex', gap: '8px' },
+  // give our textbox a border
+  input: {
+    borderWidth: 1,
+  },
+});
+
+export default () => {
+  const [tasks, setTasks] = useState<string[]>([]);
+  const [input, setInput] = useState('');
+  return (
+    <View style={styles.container}>
+      <Text>Task List</Text>
+      {/* flatlist is a canonical way to render a list of items instead of using
+      map like you would in regular react */}
+      <FlatList data={tasks} renderItem={(info) => <Text>{info.item}</Text>} />
+      <TextInput
+        style={styles.input}
+        onChangeText={setInput}
+        value={input}
+      ></TextInput>
+      <Button
+        title="Add"
+        onPress={() => {
+          setTasks((tasks) => [...tasks, input]);
+          setInput('');
+        }}
+      />
+    </View>
+  );
+};
+```
+
+Once you `yarn start`, you should be taken to Expo where you can view your changes on your browser. On the bottom left, there should also be a QR code. If you download the Expo app and scan the QR code with your phone camera ([Android](https://play.google.com/store/apps/details?id=host.exp.exponent&hl=en_US), [iOS](https://apps.apple.com/us/app/expo-client/id982107779)), it should take you to the Expo app where you can see your application in mobile form!
+
+### Learn more
+
+This was a very cursory introduction to React Native. You can learn more by referring to React docs.
+
+React Docs: [https://reactnative.dev/docs/getting-started](https://reactnative.dev/docs/getting-started)
+
+Expo Docs: [https://docs.expo.io/](https://docs.expo.io/)
+
+### Extras - Prettifying your UI
+
+[Material UI](https://mui.com/) is a library that ships a bunch of pre-style and customizable components for you to use in your own React projects. Material UI comes with a variety of components ranging from [buttons](https://mui.com/components/buttons/) to [icons](https://mui.com/components/material-icons/) to [drawers](https://mui.com/components/drawers/), all of which are customizable and come with their own props that can do common tweaks to their existing components to fit your individual need.
+
+[Semantic UI](https://semantic-ui.com/) and [Bootstrap](https://getbootstrap.com/) are two other popular libraries used for styling and do not require projects written in React. Both of them come with pre-written classes for styling elements and organizing layout.
```

Comment:
Probably would be nice to include the "React versions" of these - https://react.semantic-ui.com/, https://react-bootstrap.github.io/